Docs: Tenant attributes and collections improvements (#70179)
metabase/metabase: docs/embedding/tenants.md

![Edit tenant](./images/edit-tenant.png)

There are three ways to add a tenant attribute, using the Metabase UI, the API, or JWT SSO.

To create a tenant attribute manually using the Metabase UI:

1. Go to **Admin settings > People**
2. Select **Tenants** on the left sidebar.
3. Click on **three dots** next to the tenant.
4. Input the attribute key and value.

For details on how to set tenant attributes using JWT tenant claims, see [Setting tenant attributes using tenant claims](#setting-tenant-attributes-using-tenant-claims) below.

To set permissions based on attributes:
- Go to Permissions > Tenant collections > Root tenant collections
- You can now assign curate, view or block access to any group

> Tenant attributes will be persisted in new tenants created from JWT and existing ones, if the tenant attribute key is new. If the tenant already exists, the new tenant attribute is created if it does not exist yet. If it already existed, its value is not updated, however.

Author: I've moved it, removed the first sentence (which seems to be redundant to me), and made a small edit for clarification,

![Edit tenant](./images/edit-tenant.png)

There are two ways to add a tenant attribute. One is to create it manually using the Metabase UI:

1. Go to **Admin settings > People**
2. Select **Tenants** on the left sidebar.
3. Click on **three dots** next to the tenant.
4. Input the attribute key and value.

> The second method is to create tenant attributes from JWT SSO. Modify the JWT sign function to include a tenant attribute:

Reviewer: The sign function as such may or not exist depending what you are using. So I suggest to talk about JWT claims, which is the lingo for this:

```suggestion
The second method is to create tenant attributes from JWT SSO. In the JWT, include a claim `@tenant.attributes` to include one or more tenant attributes:
```

> The second method is to create tenant attributes from JWT SSO. Modify the JWT sign function to include a tenant attribute:

Author: Done, thanks!

2. If the tenant doesn't exist, Metabase automatically creates it. Metabase will use the value of the `@tenant` key (or your chosen assignment attribute) as the tenant slug.
3. New users are automatically assigned to the tenant from their JWT.

### Setting tenant attributes using tenant claims

To create tenant attributes from, include a claim `@tenant.attributes` to add one or more tenant attributes:

```json
{
  "@tenant": "my-tenant",
  "@tenant.attributes": {
    "some-attribute": "here it is"
  },
  "email": "another@another.com",
  "first_name": "Tester",
  "last_name": "Testerson"
}
```

> If the tenant already exists, the new tenant attribute is created if it does not exist yet. If the attribute already existed, its value is not updated, however.

Reviewer 2: > If the tenant already exists, the new tenant attribute is created if it does not exist yet

Another sentence with weird structure: `If X, then Y if Z`. Which `if` refers to what? which one should I be reading?

Do we even need "If the tenant already exists" here?

```suggestion
If a tenant attribute with this name doesn't exist, Metabase will create the attribute and assign the value from the JWT claim. However, if the tenant attribute already exists, Metabase will **not** update the value.
```

- Metabase Admins will have **Curate** access to all shared collections and all tenant collections.
- Other internal groups can be granted **View** or **Curate** access to **shared collections**, see [Configuring shared collection permissions](#configuring-shared-collections-permissions).

> - By default, non-admin internal users have **No** access to tenant-specific collections. This can be configured to allow them **Curate** or **View** access instead.

Reviewer 2: This is better just folded into the previous bullet point
